doc/contributors_guide.rst

```diff
@@ -0,0 +1,45 @@
+*******************
+Contributor's guide
+*******************
+
+Subtrees
+========
+
+are available as subtrees under ``$repo/external``.
+
+Updating the subtrees
++++++++++++++++++++++
+
+If you got a Pull Request merged in e.g. :mod:`devlib` and want to use some of
+the features you introduced in LISA, you'll need to update the subtrees. There is
+a handy LISA shell command available for that:
+
+  >>> lisa-update-subtrees
+
+This will update every subtree in the repository with the right incantation, and
+the result can be pushed straight away to LISA as a Pull Request (or included in
+a broader Pull Request).
+
+Submitting your subtree changes
++++++++++++++++++++++++++++++++
+
+Our changes to subtrees are often developped conjointly with LISA, so we write our
+modifications directly in the subtrees. You can commit these changes in the LISA
+repository, then shape those modifications into a git history ready to be pushed
+to e.g. :mod:`devlib` by running::
+
+  git subtree split --prefix=external/devlib -b my-devlib-feature
```

Comment:
following the issue I faced yesterday, that command should be run first:
```
git fetch devlib-remote
```
Otherwise, `git subtree split` will be unable to find the commit on which it is supposed to rebase your commit series.

Reply:
When we update the subtree, we also do the --fetch. But that will only help the person that updated the subtree. All other developers that want to push a change to devlib will face the issue, unless they update devlib first (which is a sensible thing to do, but is not mandatory).

Reply:
Ah gotcha